### ISSUE TITLE
AzureRMR interface

### ISSUE THREAD

**Issue by Hong-Revo** [Azure/doAzureParallel#345]
@brnleehng Would you be interested in a PR to add an [AzureRMR](https://github.com/cloudyr/AzureRMR) interface, now that it's on CRAN? This should simplify the process of setup, and also allow possible future integration with [AzureStor](https://github.com/cloudyr/AzureStor) for storage accounts.

**Comment by brnleehng** [comment]
Hi Hong, 

Are you thinking of a PR for Batch management APIs to AzureRMR?

Thanks,
Brian

**Comment by Hong-Revo** [comment]
No, I mean adding an AzureRMR interface to doAzureParallel and/or rAzureBatch. AzureRMR provides the framework for interacting with Resource Manager, which can be extended and customised by other packages.

This means people could do something like

```r
library(rAzureBatch)
az <- AzureRMR::get_azure_login()
res <- az$
    get_subscription("sub_id")$
    get_resource_group("resgrp")$
    create_batch("clusname", nodes=10, size="Standard_DS2_v2", ...etc...)

# get the batch cluster
clus <- res$get_cluster()
clus$start()
clus$stop()
# etc
```

The closest analogue might be the AzureContainers package, which has methods for creating and working with AKS clusters. See the code [here](https://github.com/cloudyr/AzureContainers/blob/master/R/add_aks_methods.R) for the methods to create the cluster. There is a [vignette](https://cran.r-project.org/web/packages/AzureContainers/vignettes/vig01_plumber_deploy.html) that describes how you would interact with the cluster (scroll down to the "deploy to a Kubernetes cluster" section).

**Comment by Hong-Revo** [comment]
Here is a reference arch repo that demonstrates using AKS to provide a real-time scoring service in R.

https://github.com/Azure/RealtimeRDeployment
